src/main/scala/no/ndla/searchapi/service/search/SearchConverterService.scala

```diff
@@ -987,9 +1008,13 @@ trait SearchConverterService {
     private def getTopicTaxonomyContexts(topic: Topic,
                                          filterVisibles: Boolean,
                                          bundle: TaxonomyBundle): Try[List[SearchableTaxonomyContext]] = {
-      val topicsConnections = bundle.topicResourceConnections.filter(_.resourceId == topic.id)
-      val topics = bundle.topics.filter(topic => topicsConnections.map(_.topicid).contains(topic.id)) :+ topic
-      val parentTopicsAndPaths = topics.flatMap(t => getParentTopicsAndPaths(t, bundle, List(t.id)))
+      val parentTopicsConnections = bundle.topicSubtopicConnections.filter(_.subtopicid == topic.id)
+      val parentTopicsAndPaths = getParentTopicsAndPaths(topic, bundle, List(topic.id))
+
+      val relevanceIds = parentTopicsConnections.length match {
+        case 0 => bundle.subjectTopicConnections.filter(_.topicid == topic.id).map(tc => tc.relevanceId.getOrElse("urn:relevance:core"))
+        case _ => parentTopicsConnections.map(tc => tc.relevanceId.getOrElse("urn:relevance:core"))
+      }
```

Comment:
For å plukke korrekt kobingsobjekt for topic måtte eg enten hente fra topic- eller subject-connections avhengig av kvar i treet emnet er. Derfor denne sjekken på lengda til parentTopicconnetions.